python/rmm/_lib/device_buffer.pyx

```diff
@@ -133,6 +133,35 @@ cdef class DeviceBuffer:
         }
         return intf
 
+    def copy(self):
+        """Returns a deep copy of DeviceBuffer.
```

Comment:
Maybe not "deep" as @jakirkham pointed out below. Also perhaps this means we need `__copy__` and not `__deepcopy__`? I don't know if `copy.copy` and `copy.deepcopy` delegate to the other dunder-method if one is not defined.
```suggestion
        """Returns a copy of DeviceBuffer.
```

Reply:
@bdice Looks like `copy.deepcopy` seems to fallback to `copy` dunder method if `__deepcopy__` isn't defined.